lms/djangoapps/mobile_api/test_middleware.py

```diff
@@ -0,0 +1,139 @@
+            platform="ios",
+            version="2.2.2",
+            expire_at=datetime(2014, 01, 01, tzinfo=UTC),
+            enabled=True
+        ).save()
+        AppVersionConfig(
+            platform="ios",
+            version="4.4.4",
+            expire_at=datetime(9000, 01, 01, tzinfo=UTC),
+            enabled=True
+        ).save()
+        AppVersionConfig(platform="ios", version="6.6.6", expire_at=None, enabled=True).save()
+
+        AppVersionConfig(platform="android", version="1.1.1", expire_at=None, enabled=True).save()
+        AppVersionConfig(
+            platform="android",
+            version="2.2.2",
+            expire_at=datetime(2014, 01, 01, tzinfo=UTC),
+            enabled=True
+        ).save()
+        AppVersionConfig(
+            platform="android",
+            version="4.4.4",
+            expire_at=datetime(9000, 01, 01, tzinfo=UTC),
+            enabled=True
+        ).save()
+        AppVersionConfig(platform="android", version="6.6.6", expire_at=None, enabled=True).save()
+
+    @ddt.data(
+        ("Mozilla/5.0 (Linux; Android 5.1; Nexus 5 Build/LMY47I; wv) AppleWebKit/537.36 (KHTML, like Gecko) "
+         "Version/4.0 Chrome/47.0.2526.100 Mobile Safari/537.36 edX/org.edx.mobile/2.0.0"),
+        ("Mozilla/5.0 (iPhone; CPU iPhone OS 9_2 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) "
+         "Mobile/13C75 edX/org.edx.mobile/2.2.1"),
+        ("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 "
+         "Safari/537.36"),
+    )
+    def test_non_mobile_app_requests(self, user_agent):
+        fake_request = HttpRequest()
+        fake_request.META['HTTP_USER_AGENT'] = user_agent
+        self.assertIsNone(self.middleware.process_request(fake_request))
+        fake_response = HttpResponse()
+        response = self.middleware.process_response(fake_request, fake_response)
+        self.assertEquals(200, response.status_code)
+        with self.assertRaises(KeyError):
+            response[AppVersionUpgrade.LATEST_VERSION_HEADER]
+        with self.assertRaises(KeyError):
+            response[AppVersionUpgrade.UPGRADE_DEADLINE_HEADER]
+
+    @ddt.data(
+        "edX/org.edx.mobile (6.6.6; OS Version 9.2 (Build 13C75))",
+        "edX/org.edx.mobile (7.7.7; OS Version 9.2 (Build 13C75))",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/6.6.6",
+        "Dalvik/2.1.0 (Linux; U; Android 5.1; Nexus 5 Build/LMY47I) edX/org.edx.mobile/7.7.7",
+    )
+    def test_no_update(self, user_agent):
+        fake_request = HttpRequest()
+        fake_request.META['HTTP_USER_AGENT'] = user_agent
+        self.assertIsNone(self.middleware.process_request(fake_request))
+        fake_response = HttpResponse()
+        response = self.middleware.process_response(fake_request, fake_response)
+        self.assertEquals(200, response.status_code)
+        with self.assertRaises(KeyError):
```

Comment:
`self.assertNotIn(AppVersionUpgrade.LATEST_VERSION_HEADER, response)` is more idiomatic.